comms/dht/src/crypt.rs

```diff
@@ -102,7 +104,7 @@ pub fn create_message_challenge(header: &DhtMessageHeader, body: &[u8]) -> [u8;
 }
 
 /// Generates a 32-byte hashed challenge that commits to all message parts
-pub fn create_message_challenge_parts(
+pub fn create_message_domain_separated_hash_parts(
```

Comment:
```suggestion
pub fn create_message_hash(
```
The previous name conveys the intent well enough, but the addition of `_parts` is confusing, so `create_message_challenge` could also work.